agent/src/agent/monitoring/streamsets.py

```diff
@@ -28,8 +29,7 @@ def _is_influx(pipeline_: pipeline.Pipeline):
     return pipeline_.source.type == source.TYPE_INFLUX
 
 
-def _pull_pipeline_metrics(pipeline_: pipeline.Pipeline):
-    jmx = sdc_client.get_jmx(pipeline_.streamsets, f'metrics:name=sdc.pipeline.{pipeline_.name}.*')
+def _pull_pipeline_metrics(pipeline_: pipeline.Pipeline, jmx=None):
```

Comment:
I guess jmx is not an optional parameter now?
for bean in jmx['beans'] will raise TypeError in case of None.